src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -417,8 +417,9 @@ static bool process_pending(CompileTask* task) {
 
 void CompileQueue::transfer_pending() {
   assert(_lock->owned_by_self(), "must own lock");
-  while (!_queue.empty()) {
-    CompileTask* task = _queue.pop();
+
+  CompileTask* task;
```

Comment:
I assume these changes aren't strictly needed (`_queue` is always drained under the lock).